libr/parse/code.c

```diff
@@ -54,3 +16,304 @@ R_API int r_parse_is_c_file (const char *file) {
 	}
 	return false;
 }
+char *racp_get_next_token_dualbuf(RDualBuf *b, int *type) { //lexical analyser, finite state machine simulation
+	char *start = r_dualbuf_current_charp (b), *current = start, c = 0;
+	int status = 0,  counter = 0;
+	char *reservedwords[] = {"struct", "enum", "union", "typedef",NULL};
+	while (*r_dualbuf_current_charp (b) != EOF) {
+		//lets parse the whole thing
+		if (counter++ >= 1024) {
```

Comment:
use ++counter since if counter is 1023, will pass since the increment will be after checking against 1024